src/Magnum/Platform/EmscriptenApplication.cpp

```diff
@@ -0,0 +1,505 @@
+
+    auto mouseMoveCallback =
+        [](int, const EmscriptenMouseEvent* event, void* userData) -> Int {
+            MouseMoveEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseMoveEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_mousemove_callback("#canvas", this, false, mouseMoveCallback);
+
+    auto wheelCallback =
+        [](int, const EmscriptenWheelEvent* event, void* userData) -> Int {
+            MouseScrollEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseScrollEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_wheel_callback("#canvas", this, false, wheelCallback);
+
+    // TODO: Trying to map touch to mouse opened quite a can of worms:
+    //  1. Tap is already mapped via click by emscripten => mouse press, mouse move, mouse release
+    //  2. Touch down, move, up is not a tap
+    //  3. If single touch, everything is clear, second touch-down could be handled with canceling, but
+    //     would require a manual mouse release event. Or we could continue just following the first touch?
+    //
+    // Map first touch as left mouse, yes or no?
+    /*
+    emscripten_set_touchstart_callback("#canvas", this, false,
+        [](int, const EmscriptenTouchEvent* event, void* userData) -> Int {
+            MouseEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mousePressEvent(e);
+            return e.isAccepted();
+        });
+    emscripten_set_touchend_callback("#canvas", this, false,
+        [](int, const EmscriptenTouchEvent* event, void* userData) -> Int {
+            MouseEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseReleaseEvent(e);
+            return e.isAccepted();
+        });
+    emscripten_set_touchmove_callback("#canvas", this, false,
+        [](int, const EmscriptenTouchEvent* event, void* userData) -> Int {
+            MouseMoveEvent e{event};
+            reinterpret_cast<EmscriptenApplication*>(userData)->mouseMoveEvent(e);
+            return e.isAccepted();
+        });
+    */
+
+    /* keypress_callback does not fire for most of the keys and the modifiers don't seem to work,
+      keydown on the other hand works fine for all */
+    auto keyDownCallback =
+        [](int, const EmscriptenKeyboardEvent* event, void* userData) -> Int {
+            EmscriptenApplication* app = reinterpret_cast<EmscriptenApplication*>(userData);
+            if(app->isTextInputActive() && strlen(event->key) == 1) {
+                TextInputEvent e{{event->key, 1}};
+                app->textInputEvent(e);
+                return e.isAccepted();
+            }
+            KeyEvent e{event};
+            app->keyPressEvent(e);
+            return e.isAccepted();
+        };
+    emscripten_set_keydown_callback("#canvas", this, false, keyDownCallback);
```

Comment:
what about doing directly

```suggestion
    emscripten_set_keydown_callback("#canvas", this, false, [](int, const EmscriptenKeyboardEvent* ...) {
```

instead? Similarly elsewhere.